crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -659,3 +610,16 @@ pub enum DiagnosticFormat {
     /// This may use color when printing to a `tty`.
     Concise,
 }
+
+/// Creates a `Diagnostic` from a parse error.
+///
+/// This should _probably_ be a method on `ruff_python_parser::ParseError`, but
+/// at time of writing, `ruff_db` depends on `ruff_python_parser` instead of
+/// the other way around. And since we want to do this conversion in a couple
+/// places, it makes sense to centralize it _somewhere_. So it's here for now.
+pub fn create_parse_diagnostic(file: File, err: &ruff_python_parser::ParseError) -> Diagnostic {
```

Comment:
A use case for your `IntoDiagnostic` trait ;)

Reply:
Hmmm. I don't _think_ this on its own asks for generics. If it weren't for the dependency graph (which we briefly talked about fixing at the on-site), then this could "just" be a concrete method on `ParseError` and things would be fine.